src/sentry/integrations/github/webhook.py

```diff
@@ -356,26 +369,36 @@ def _handle(
                         date_added=parse_date(commit["timestamp"]).astimezone(timezone.utc),
                     )
                     for fname in commit["added"]:
+                        languages.add(get_file_language(fname))
                         CommitFileChange.objects.create(
                             organization_id=organization.id,
                             commit=c,
                             filename=fname,
                             type="A",
                         )
                     for fname in commit["removed"]:
+                        languages.add(get_file_language(fname))
                         CommitFileChange.objects.create(
                             organization_id=organization.id,
                             commit=c,
                             filename=fname,
                             type="D",
                         )
                     for fname in commit["modified"]:
+                        languages.add(get_file_language(fname))
                         CommitFileChange.objects.create(
                             organization_id=organization.id,
                             commit=c,
                             filename=fname,
                             type="M",
                         )
+
+                    languages.discard(None)
+                    repo_languages = set(repo.languages)
+                    repo_languages.update(languages)
+                    repo.languages = list(repo_languages)
```

Comment:
maybe make this shorter?
```suggestion
                   repo.languages = list(set(repo.languages).union(languages))
```

Also if the db value for languages is initially null, `set(repo.languages)` will error. Is this a concern?